src/journal/src/Journal/Internal/Utils.hs

```diff
@@ -37,9 +38,18 @@ withRWFd fp k =
 int2Int32 :: Int -> Int32
 int2Int32 i@(I# i#) = assert (i <= fromIntegral (maxBound :: Int32)) (I32# i#)
 
+int322Int :: Int32 -> Int
+int322Int (I32# i#) = assert (I# i# <= (maxBound :: Int)) (I# i#)
+
```

Comment:
Hmm these `assert` looks a bit funny, to have here but sure why not (would maybe be cool to have as a static_assert but I don't think Haskell has those. Btw why not check `minBound` as well? These are Ints after all.

Reply:
Good point, will add `minBound` checks also tomorrow.